### ISSUE TITLE
Switch godep to go modules?

### ISSUE THREAD

**Issue by feiskyer** [kubernetes/autoscaler#2042]
Kubernetes has moved from godep to go modules, I think we should do same for autoscaler.

However, the dependency overrides are still required for some packages. Not sure whether there are good solutions for this, @mwielgus @maciekpytel @bskiba any ideas on this?

**Comment by mboersma** [comment]
You should be able to use the `replace` directive to keep using the overrides:

```
replace github.com/Azure/go-autorest => ./_override/github.com/Azure/go-autorest
```

But I've been experimenting with converting cluster-autoscaler to go modules and I don't think this go modules feature is working (or else I'm doing something wrong).

**Comment by feiskyer** [comment]
Already fixed by #2102

/close
